### ISSUE TITLE
Error on Windows10(x64)

### ISSUE THREAD

**Issue by NinaNVY** [dflynn15/gulp-jasmine-phantom#53]
![219](https://cloud.githubusercontent.com/assets/15184314/12783284/9103172a-ca91-11e5-9a8f-cc1654ff27b6.png)
Hi, 
I have PhantomJS installed globally. Phantomjs binary is available in the PATH and executable from the command line. 
Here is my gulp task
```
gulp.task('tests', function () {
	return gulp.src('tests/spec/extraSpec.js')
		.pipe(jasmine({
			integration: true
		}));
});
```
I can not find logs and events.js:141 to see what's going wrong.
Running Jasmine with minijasminenode2 goes well when I delete the string `integration: true` from my gulp task
Please, help

**Comment by dflynn15** [comment]
Would you mind sharing your spec file?

**Comment by NinaNVY** [comment]
Yes of course :)
I'm taking Udacity "Web Tooling & Automation", they gave this spec:
```
describe('window height', function() {
	it('returns window height', function() {
		expect(getWindowHeight()).toEqual(jasmine.any(Number));
	});
});
```
And the whole code is here: [https://github.com/udacity/ud892/tree/master/Lesson%204]

Thank you!

**Comment by dflynn15** [comment]
So, I was going through the code given and noticed that your gulp task seems to be missing the `  vendor: 'js/**/*.js'` found here: https://github.com/udacity/ud892/blob/master/Lesson%204/gulpfile.js#L46

That may be the source of the problem. Let me know if that helps!

**Comment by NinaNVY** [comment]
Thank you for your time dear Daniel! 
My code already has this `vendor: 'js/**/*.js'` at line 46 :( 
To detect Error I tried to run:
`gulp > gulp_log.log 2> gulp_err.log`
The result is here:
![239](https://cloud.githubusercontent.com/assets/15184314/12913221/74565e60-cf2e-11e5-93dc-f56de4d1ede4.png)
Maybe we could cut this problem into smaller pieces, something like this:
First, install and test jasmine alone (as node-module)
  if it will work, then test jasmine with phantomjs to search for a problem with compatibility
      if it will work, then search for a problem with gulp
Does it make sense?

**Comment by dflynn15** [comment]
I went ahead and cloned the courses repository down and everything works fine after `npm install` for all of the missing packages (they seriously need to give you guys a `package.json` file).

I think a good course of action to determine where the error might be coming from is to include this option in the task: `keepRunner`. This will allow you to have a spec runner html file after it breaks. After you run the task again `gulp tests` you should see a generated `specRunner.html` file that you can open and then determine if the error is in your tests.

**Comment by NinaNVY** [comment]
Thank you!
I added `keepRunner: true;`
After I ran the task again, a new spec runner html file didn't appear. I just opened the old specRunner.html, given by Udacity, and it showed no error.
![246](https://cloud.githubusercontent.com/assets/15184314/12946085/dea59644-d002-11e5-9ccc-1960899f8f92.png)
I want to `npm install`  all of the missing packages, as you did. Would you tell me which commands to run?

**Comment by dflynn15** [comment]
@NinaNVY. I totally lead you astray. It's `keepRunner: './'` not `keepRunner: true` :sob:

**Comment by NinaNVY** [comment]
LOL :)
Thank you! Fixed `keepRunner: './'` :) and now it finally creates the new `runner` html file, which shows no errors :)
But after I ran `gulp tests` task, I waited 3 and a half of hours :) Here is the state of cmd all this time and the new error file 
![249](https://cloud.githubusercontent.com/assets/15184314/13013047/cd6060d6-d1be-11e5-9054-4850739169c1.png)

`Gulp tests `task created new  `runner`, but after that it is simply hanging on and sleeping. So, after a few hours I pressed `Ctrl+C` :)
In case if it's possible that gulp-jasmine-phantom install got messed up I decided to try running:
```
npm uninstall gulp-jasmine-phantom
npm install -g jasmine-core
npm install -g phantomjs
npm install gulp-jasmine-phantom
```
Which didn't help

**Comment by dflynn15** [comment]
Well that's infuriating. Could you give me your system specs (OS, Node version, NPM version, and anything else you want to add)? I will try to set up a VM and see what I can do from there.

**Comment by NinaNVY** [comment]
Yes, of course, but isn't it too much work to create new VM? Maybe it would be easier to use TeamViewer?

**Comment by NinaNVY** [comment]
I tried run the following command:
`phantomjs /path/to/run-jasmine.js http://localhost/js/test/unit/`
, here is the output:
![280](https://cloud.githubusercontent.com/assets/15184314/13028110/3938de16-d276-11e5-8560-5d22ef718d93.png)

after that it is simply hanging on and sleeping. Maybe this is the reason?

**Comment by dflynn15** [comment]
That's super interesting. Have you ever been able to run `phantomjs` normally without running into that issue? 

For example saving this as a tiny little JS file and running it: 
```js
console.log('Loading a web page');
var page = require('webpage').create();
var url = 'http://phantomjs.org/';
page.open(url, function (status) {
  //Page is loaded!
  phantom.exit();
});
```

**Comment by NinaNVY** [comment]
It printed out all console.log, but didn't open any browser and the given page
![281](https://cloud.githubusercontent.com/assets/15184314/13083401/e4d029de-d4e5-11e5-970e-0bdd7e2fe1b0.png)

**Comment by dflynn15** [comment]
Well, that's actually good. Concerning for me, but it means that your `phantomjs` works! I'll spin up that VM and start working on it. I know that there are some `phantomjs` issues with Windows, but that shouldn't be causing that issue.

**Comment by NinaNVY** [comment]
Hi Daniel!
Thank you!
Would you peek at what happened after I ran `npm install path` ?
Here is a brand new message from phantomjs :) 
![282](https://cloud.githubusercontent.com/assets/15184314/13140568/ea0e68de-d643-11e5-8489-1182976b860a.png)

**Comment by dflynn15** [comment]
Ok! So. I went ahead and forked that Lesson repository so that we help narrow down specific issues pertaining to that code base.

You can go on over [here](https://github.com/dflynn15/ud892) and clone that repository. Head on over to the `Lesson 4` directory and do an `npm install`. After that, run `gulp tests` and let's see if this helped or not!

**Comment by NinaNVY** [comment]
Hi Daniel!
I cloned your fork with the new JSON package. Thank you! :smile:  :+1: 
> and do an `npm install` 

but I have all these node modules already installed, should I install again?

![247](https://cloud.githubusercontent.com/assets/15184314/13157236/c4c15a9c-d697-11e5-9af9-7d41eec49743.png)

**Comment by dflynn15** [comment]
Yep! There may be version differences and many other things. If you feel ok with it, I would encourage you do run `npm cache clean` before `npm install` just to ensure that all of the packages are installed cleanly and are up to date.

**Comment by NinaNVY** [comment]
>  I would encourage you do run `npm cache clean` before `npm install`

I don't know what will be cleared in npm cache, if not too much trouble could you tell me in short? I know you are busy I just need some time to google it (sorry I'm a newbie).
Should I run `npm uninstall` old node modules before `install` new version?

**Comment by dflynn15** [comment]
Here is the definition from the `npm` docs:
>clean: Delete data out of the cache folder. If an argument is provided, then it specifies a subpath to delete. If no argument is provided, then the entire cache is cleared.

What that means is that when you run `npm install` a cached version of that module is stored on your computer so that `npm` doesn't need to make extra network requests when you make subsequent `npm install` on other projects that use the same modules.

`npm cache clean` will wipe that away so that you will _definitively_ use the versions explicitly described in the `package.json`

**Comment by NinaNVY** [comment]
Thank you for the explanation! :+1: 
First I `uninstall` all the old modules and second - clear the npm cache? Then install new.

**Comment by henrikrudstrom** [comment]
I have the same problem on  windows, that is , when i run the gulp task it hangs with the text "Running Jasmine with Phantom"

could it be realated to this?

>##  Spawning .bat and .cmd files on Windows
The importance of the distinction between child_process.exec() and child_process.execFile() can vary based on platform. On Unix-type operating systems (Unix, Linux, OSX) child_process.execFile() can be more efficient because it does not spawn a shell. On Windows, however, .bat and .cmd files are not executable on their own without a terminal and therefore cannot be launched using child_process.execFile() (or even child_process.spawn()). When running on Windows, .bat and .cmd files can only be invoked using either child_process.exec() or by spawning cmd.exe and passing the .bat or .cmd file as an argument (which is what child_process.exec() does). 

[from the node docs](https://nodejs.org/api/child_process.html#child_process_asynchronous_process_creation)

I tried replacing execFile with exec at at least in doesnt hang... will see if i get time to investigate further

**Comment by henrikrudstrom** [comment]
actually, still same behaviour with ´exec´ as with ´execFile´

**Comment by NinaNVY** [comment]
@henrikrudstrom  please look at the end - when I tried to run phantomjs with jasmine, it asks for process variable
![284](https://cloud.githubusercontent.com/assets/15184314/13177580/22374e32-d72b-11e5-87aa-cce644b9c4db.png)

**Comment by NinaNVY** [comment]
@dflynn15 I can not install gulp and gulp-eslint
![299](https://cloud.githubusercontent.com/assets/15184314/13179047/118e913c-d733-11e5-880f-68f686becc52.png)
![298](https://cloud.githubusercontent.com/assets/15184314/13179075/419e5196-d733-11e5-9617-c5449c24a5c4.png)
![294](https://cloud.githubusercontent.com/assets/15184314/13179096/5c6bdc46-d733-11e5-8f6b-2b3090a2ebf5.png)

**Comment by NinaNVY** [comment]
@dflynn15 
Hi, Daniel!
I need more time to follow your advice, sorry, because I have trouble now even installing Gulp. Gulp Installation tells me again and again about graceful-fs :cry: 
 I installed the last '4.1.3' version of graceful-fs , but Gulp Installation tells me again about graceful-fs !
![300](https://cloud.githubusercontent.com/assets/15184314/13181110/754fffb8-d73c-11e5-8c9a-098b36fca194.png)

Here is what the npm member **ashleygwilliams** [answered](https://github.com/npm/npm/issues/11607#issuecomment-186272179):
"probably graceful-fs is a dependency of a dependency that you are using and the semver expression that dep is using does not allow version 4. you'll have to look at the package.json of the dep that requires it. this is also why u see the lodash warning and the second graceful-fs warning.
updating your other dependencies may help fix this as those lib maintainers likely (hopefully!) updated those deps."

Sorry, I'm a newbie and I need more time to understand what she wrote :(

**Comment by dflynn15** [comment]
@NinaNVY, so first things first. Let's talk about the error when installing `gulp` globally. The error there is simply saying that you do not have permission to install a global package. Usually that means the way you installed `node` and `npm` was through a `sudo` or the directories and libraries do not give you read/write access. 

A quick way around that is to `sudo npm install -g gulp` but ideally you should figure out how to have read/write access to those folders. I'll link later once I find the SO answer I used a long time ago.

As for the warnings, that's just simply saying that there are dependencies that are "out of date." This is not necessarily a bad thing. Some modules depend on specific versions because the community as a whole doesn't follow [semver](http://semver.org/) very well. So on occasion you'll see modules explicitly depend on older versions. In this case there are dependecies that are looking for older versions of `lodash` and `graceful-fs`. This isn't a problem, and the warning is simply letting you know, but not erroring out :+1:

**Comment by NinaNVY** [comment]
@dflynn15 

Oh, Daniel, thank you for your time and kind attention! :)

> so first things first. Let's talk about the error when installing gulp globally

Did I install gulp globally if I ran `npm install --save-dev gulp` ?

Would you tell me please how can I install `gulp-jasmine-phantom: 3.0.0-rc1 `? I can only install version 2.1.2

**Comment by dflynn15** [comment]
> Did I install gulp globally if I ran npm install --save-dev gulp ?

So here's a quick rundown on the install flags and what each means. _This is taken from the `npm install` [docs here](https://docs.npmjs.com/cli/install).

npm install takes 3 exclusive, optional flags which save or update the package version in your main package.json:
- `-S, --save`: Package will appear in your dependencies.
- `-D, --save-dev`: Package will appear in your devDependencies.
- `-O, --save-optional`: Package will appear in your optionalDependencies.

When using any of the above options to save dependencies to your package.json, there is an additional, optional flag:
- `-E, --save-exact`: Saved dependencies will be configured with an exact version rather than using npm's default semver range operator.

The -g or --global argument will cause npm to install the package globally rather than locally. See [npm-folders](https://docs.npmjs.com/files/folders).


All in all, when you did `--save-dev` what it did was install a local copy of `gulp` which is totally fine. However, you won't be able to use `gulp` outside of the project.

As for the version `3.0.0-rc1` it will be installed if you run `npm install` from the directory containing the `package.json` ([see here](https://github.com/dflynn15/ud892/blob/master/Lesson%204/package.json#L19)).

I think it would benefit you a lot to go through `npm`s documentation and learn how it works. Specifically what a `package.json` file is, how it defines packages, and what `npm` tasks such as `install` or `start` do.

**Comment by NinaNVY** [comment]
Dear Daniel!
Thank you for your explanation and for your time. Very very much appreshiated. Thank you!
I always read documentation carefully. In the case of Udacity "Web Tooling & Automation" course they gave us just a quick overview: 1) please install Node, Gulp, Sass, Autoprefixer, Browsersync, ESlint, Phantom, Gulp-Jasmine-Phantom, Babel, Source map etc.
2) Watch the video and try yourself how it works, to give us the taste.
3) Spend more time to read documentation for each tool carefully.
On step 2 I tried every tool, except of gulp-jasmine-phantom. I saw how it works only in the instructor's video. And perhaps it make sence to move to step 3, and after completing step 3, go back to step 2 gulp-jasmine-phantom, because I'm wasting your time, sorry for the trouble.
It'll take me some time to read all the documentation, and after that go back to this issue.
Here is my intermediate result after reinstallation, it hangs on and sleeps :(
![301](https://cloud.githubusercontent.com/assets/15184314/13196069/f139ec72-d7d4-11e5-924a-11fa68a8dd85.png)

**Comment by andretf** [comment]
Same error here on Linux.
I get stuck on "Running Jasmine with PhantomJS".
I called my specRunner.html with `page.render` from `phantomjs` and got this screen:

![a](https://cloud.githubusercontent.com/assets/5576877/13355850/21dc9310-dc81-11e5-934a-b07f3861e5a0.png)

**Comment by dflynn15** [comment]
Please post those specs so I can help diagnose the problem. There isn't much I can do to debug the issue with just a screenshot 😕

**Comment by andretf** [comment]
Nevermind, problem was on my side. Sorry I didn't read carefully docs.
I was running a custom `specHtml` which **didn't console messages**<sup>[1]</sup>.
So, `phantomjs` on `jasmine-runner.js` **never exited**.
I didn't test, but I'm sure that it will work if you put some `console.log(finishRegEx)` to execute after when jasmine finishes tests, just in case...
I don't know if it's really a good idea, but at least I'd suggest please to comment about this behavior in docs.
Anyway, thank you so much for your hard work on this project!

<sup>1</sup>:  because I didn't think using `vendor`. Somehow, I assumed it was just for 3rd party libraries, I think that `include` would be a name that would fit better to this property.

**Comment by alexkonkin** [comment]
I have got the same error in my environment (Window 7, x64)
I just cloned this repository (udacity), added all packages that were required by
the test suite and got the error that was mentioned in that thread.

How it could be fixed?

**Comment by alexkonkin** [comment]
http://screencast.com/t/RSB8JZYIpl

**Comment by alexkonkin** [comment]
http://screencast.com/t/IgTGAa4j

**Comment by alexkonkin** [comment]
Has anyone managed to fix this?

**Comment by alexkonkin** [comment]
To replicate this error it should be enough to:
- clone the repository https://github.com/udacity/ud892.git
- check that phantomjs.exe is present under your classpath,
  let's say c:\windows\system32
- check that phantomjs can be executed from the command line
  and run some *.js test script
- navigate to the Lesson 4 folder
- run gulp, then add all packages that are required by gulp
- run 
  gulp tests

Expected behavior:
gulp should be able to run the test suite and return the result

Actual result:
The error is shown:
[11:31:37] Finished 'tests' after 6.31 ms
[11:31:37] Running Jasmine with PhantomJS

{ [Error: ENOENT: Tests contained failures. Check logs for details.]
  message: 'ENOENT: Tests contained failures. Check logs for details.',
  showStack: false,
  showProperties: true,
  plugin: 'gulp-jasmine-phantomjs',
  __safety: { toString: [Function: bound ] } }

**Comment by alexkonkin** [comment]
I am adding my bug here  because in all comments above people describe
their environment. In this report I created a test repository and
scenario as simple as it could be. Someone who would like to
replicate this problem just need to clone a  test repo
install required packages and debug this problem

Environment:
This bug report is related to the recent version of phanotmjs - 2.1
and the one that is installed via npm (1.9.8)
Windows 7 Enterprise, x64
Nodejs: 4.4.0

Steps to reproduce:

    download a test repository with as simple test as it could be: https://github.com/alexkonkin/gulp-jasmine-phantom_bug.git
    install required plugins: npm install gulp npm install gulp-jasmine-phantom npm install phantomjs (or place phantomjs.exe binary to Windows/System_32 folder)
    run tests by executing the command gulp tests

Expected behavior:

    tests task that represents an execution of single test that engages the utilization of this plugin should be passed

Actual result:

    the test is failed with the error: [10:24:26] Running Jasmine with PhantomJS

events.js:141
throw er; // Unhandled 'error' event
^
Error: ENOENT: Tests contained failures. Check logs for details.

Please also refer to the screenshot below:
http://screencast.com/t/D7GGTFJ3qwC

Other info:
Another plugin
(https://github.com/sandermak/gulp-jasmine2-phantomjs)
is also unable to run the test. Has something changed in phantomjs.exe binary?

Jasmine2 can execute this test on its own:
http://screencast.com/t/Fn5kmjbypv

**Comment by dflynn15** [comment]
I cloned down the bug repo you provided and was able to run the tests running on my Mac.

![screenshot 2016-03-21 14 12 17](https://cloud.githubusercontent.com/assets/1205286/13929150/1a509258-ef6f-11e5-9eb8-026b4b28a7b8.png)

Hopefully tonight I can get my Windows VM running correctly, but I would not be surprised if it had something to do with the Windows build and paths.

**Comment by alexkonkin** [comment]
Thank you so much for the instant response.

As you can see from the output this test is as simple as it could be 
and was designed to check that the phantomjs.exe binary can be spawned by the software.
Currently I don't know the reason of this problem but can suppose that somethting
has been changed exactly with the windows builds of phantomjs.

**Comment by Nysosis** [comment]
I've been having a similar issue, going through the current process I think I've identified *an* issue, haven't had the time to check further before finishing for the day, but thought Id leave it here as a potential starting point if I don't get a chance to follow up.

It looks like the handlebars is rendered with absolute paths to assets i.e. `c:\projects\my-project\src\js\some-javascript.js` which isnt a valid path for the html, I would have expected `file://c:/projects....`, with the file protocol and forward slashes.

Its possible phantomjs handles this, and its a red herring on my part, but putting logging into `jasmine-runner.js` it was failing to open the file with those paths in, removing the tags made it open fine

**Comment by dflynn15** [comment]
@Nysosis if you modify the `jasmine-runner.js` locally to accept the `file://` does the task run then? I haven't looked too deeply into why Windows version would require the `file://` however the unix based systems seem to handle absolute paths in `src` fine.

**Comment by Nysosis** [comment]
I tried that, but it still gave the same `Tests contained failures` error
(I think because of backslashes instead of forward slashes). Where I got to
was taking that generated html, fixing it up, saving it elsewhere, and
passing it along in the `specHTML` config option. That stopped the `Tests
contained failures` message from happening, however nothing else was
happening either (i.e. the task didn't finish), that might be due to no
tests running or something - I had to leave at the point before being able
to look any further into it

**Comment by Nysosis** [comment]
Right, been doing some more fiddling around at home, and managed to get it running and outputting to the console as expected.

What I've had to do to get it to work:

1. [`jasmine-runner.js - line 32`](https://github.com/dflynn15/gulp-jasmine-phantom/blob/master/lib/jasmine-runner.js#L32) - Change this to be: `page.open("file:///" + system.args[1], function(status) {`
1. ['index.js - line 144`](https://github.com/dflynn15/gulp-jasmine-phantom/blob/master/index.js#L144) - Change this to be:
    
   ```
 var fixupPath = function(path) {
        if (path.match(/^http/)) {
          return path;
        }
        
        return "file:///" + path.replace(/\\/g, '/');
    }
    
    // Create the compile version of the specRunner from Handlebars
    var specData = handlebar.compile(data),
        specCompiled = specData({
          files: filePaths.map(fixupPath),
          jasmineCss: fixupPath(jasmineCss),
          jasmineJs: jasmineJs.map(fixupPath),
          vendorJs: vendorJs.map(fixupPath),
          specRunner: fixupPath(specRunner)
        });
```

This works for me on a w7 box, no idea how it would affect *nix machines running the plugin.

**Comment by NinaNVY** [comment]
@dflynn15 @Nysosis  I did your additions to the code. It prints out to to the SpecrRunner.html this information from extraSpec.js:
```
describe('window height', function() {
	it('returns window height', function() {
		expect(getWindowHeight()).toEqual(jasmine.any(Number));
	});
});
```
but doesn't log the same information to the `cmd`

![351](https://cloud.githubusercontent.com/assets/15184314/13988390/c539f610-f11c-11e5-986a-8765cd2526ae.png)

**Comment by Nysosis** [comment]
Can you run this command in the folder: `npm list -depth=0`, and put the output here, just so we can see what versions of packages you're using.

There's something else that I've noticed when working with that [workshop repo](https://github.com/udacity/ud892) (I presume that's the one you're using? I grabbed it from earlier messages in the issue), the runner doesn't like the space in the folder name 'Lesson 4'. If I try to run `gulp tests` from there, it falls over, renaming the folder to have no space makes it work fine, and outputs to the console as expected.

**Comment by NinaNVY** [comment]
@Nysosis 
I renamed the folder, but same result :(
And from this Lesson4 folder I installed only gulp and gulp-jasmine-phantom, because all the other packages I have installed as a node modules.  Sorry if it isn't right :)
![352](https://cloud.githubusercontent.com/assets/15184314/14012615/d52581a2-f1b7-11e5-8a47-26189310f574.png)

**Comment by alexkonkin** [comment]
@NinaNVY
My suggestion would be to check a test repo that I created and which does not contain any additional settings except one test that could allow us to detect if something is wrong with this plugin and phantomjs.exe binary otherwise we will start to debug this course from udacity and not the actual problem.

I started from the same course and then created test git repository and shared it in my bug report:

https://github.com/sandermak/gulp-jasmine2-phantomjs

repository does not contain any packages that means that you should install
them before testing. Such approach would allow people to check the same thing
and not the particular problems with someone's environment.

**Comment by alexkonkin** [comment]
@NinaNVY
I think that you can fork my repository (gulp-jasmine-phantom_bug.git), install packages and check it in your environment with
the workaround that was suggested, then you can commit your changes that would allow other people
to checkout it and test in their environments.

p.s.: please consider the suggestions above only as ideas that would allow us 
to boost the fixing process.

**Comment by NinaNVY** [comment]
@alexkonkin Thank you!
I did it and here is the result:
![353](https://cloud.githubusercontent.com/assets/15184314/14023986/2c6c7458-f1f8-11e5-8b44-28b8e3a5492d.png)

**Comment by alexkonkin** [comment]
Any updates on this bug?
If the problem is that no one is able to fix this because does not have access to the Windows environment
I can create a virtual machine in my home PC and share with you Remote Desktop credentials.

Please let me know if this would be acceptable for you.
Thank you.

**Comment by dflynn15** [comment]
Right now I am slammed with my job so maintaining this has been put lower on my priority list right now. You can always submit a PR with a fix and I can help validate the *nix side of things. It's much easier for me if you submit a PR rather than waiting on me to write and test the fix you proposed here.

**Comment by dflynn15** [comment]
@Nysosis I have a [branch up here](https://github.com/dflynn15/gulp-jasmine-phantom/tree/windows-phantom-path). Would you mind testing that on your Windows env?

**Comment by juliuskovac** [comment]
I had the same issue when going through that course (on Windows 10, x64).
I can confirm that after strugling with this issue for more than half a day I was able to make it work applying the changes @Nysosis commented on Mar 22 (big thank you for that!).

I also gave a try for that branch you proposed @dflynn15.
When I changed content of my gulp-jasmine-phantom with downloaded zip of that branch it doesn't work straight away failing with
`PS D:\itumb\itumb-live-fe> gulp tests
(node:264) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
[08:21:44] Using gulpfile D:\itumb\itumb-live-fe\gulpfile.js
[08:21:44] Starting 'tests'...
[08:21:44] Finished 'tests' after 7.93 ms
[08:21:44] Running Jasmine with PhantomJS
Couldn't load the page

events.js:160
      throw er; // Unhandled 'error' event
      ^
 Error: 1: Tests contained failures. Check logs for details.`

When I look closer and compare your changes to backup I made from my patched version, I found out that jasmine-runner.js was missing a patch @Nysosis suggested, something like this:
`page.open("file:///" + system.args[1], function(status) {`

I am not sure if this will work on linux/maxos but after this change that branch works for windows.

**Comment by gwynjudd** [comment]
Hello,

I found one potential issue, I was getting the same error as the original:

![image](https://cloud.githubusercontent.com/assets/314188/15304111/9b65bc02-1c0f-11e6-9dc5-3149b3aa2e08.png)


On some debugging, I found that on this line it is trying to spawn phantomjs.cmd:

https://github.com/dflynn15/gulp-jasmine-phantom/blob/master/index.js#L22

I have installed phantomjs version 2.1.1, but it doesn't come with phantomjs.cmd. There is only a .exe. Either way, you could just run it as "phantomjs" without extension and it would work.

Changing that line got me passed the "ENOENT" problem, and onto another problem, but this should be fixed anyway.

**Comment by gwynjudd** [comment]
Once I took the fix from @Nysosis (https://github.com/dflynn15/gulp-jasmine-phantom/issues/53#issuecomment-200033552), plus my fix for phantomjs.cmd, it was then working for me.

Windows 8
PhantomJS 2.1.1
NodeJS 4.4.4

**Comment by enderandpeter** [comment]
Wow... Google you should be ashamed of yourselves for giving us so much broken code in your Udacity courses. So the solution is to basically fix the damn course code?

**Comment by enderandpeter** [comment]
Please merge the commits from #66 that are shown in the comment because I too confirm that it is what allows all of this to work on Windows with that Udacity course.

**Comment by mnr1** [comment]
Any idea when this fix will be put in a release? I see that it has been merged into the master branch, but no release has been made with these updates. In my opinion, this is a pretty major issue if the plugin doesn't work on Windows computers at all.

**Comment by dflynn15** [comment]
@mnr1 I realize the concern. It has been in testing to make sure it does not break other OS through the change. Because there are other features that are being merged in as well I have opted to wait. If you need the change simply use the Github url for your `package.json` and keep an eye on this issue. Once the fix has been updated and published as a versioned plugin I'll close the issue.

For now, I just ask you please be patient with me as I deal with other job and life related issues.

**Comment by mnmercer** [comment]
@dflynn15 Thank you very much for your quick response. We are trying to use this plugin in the build of our first release of software so I was curious about the status. Attempting to use the GitHub URL right now.

UPDATE:
I tried using the the latest GitHub code, but when I run the plugin on my Windows 10 machine, it runs into an error when trying to execute phantomjs. I am using the `phantomjs-prebuilt` [node module ](https://www.npmjs.com/package/phantomjs-prebuilt) instead of a globally installed version of PhantomJS so I expected that the plugin will use that binary. However, when I did some debugging, I discovered that `exec("where phantomjs")` in `hasGlobalPhantom()` is not returning an error like it should. Instead, it is finding the phantomjs binary in the `node_modules` directory and assumes that PhantomJS is installed globally. Then when we run `execPhantom()`, it can't find the `phantomjs` command because it doesn't exist globally. I even tried changing the command to `phantomjs -v` and it returned the version of the prebuilt PhantomJS instead of failing.  Not quite sure why `exec()` is behaving differently than when I run the commands in my command prompt. Any suggestions?

UPDATE AGAIN:
I think I figured out the issue. I was running the gulp plugin through my `package.json` file and apparently, Node looks for all binaries within the `node_modules` folder. However, this does mean that it was pulling the correct binary the whole time. So even when I run the plugin straight from my command line, I still have the same issue. `execPhantom()` is failing when it tries to execute.

UPDATE AGAIN AGAIN:
I apologize about all this. Was going through old closed issues and found this one #44. The fix proposed at the end of the discussion fixed my issue. However, that means that the `phantomExecutable` needs to be the `.cmd` one if it's being run on a Windows. Is there any work on making this happen?

**Comment by dflynn15** [comment]
@mnr1 I am going to revert the change that looks for the `.cmd` executable. Also, if you have updates can you just add a reply? That way I can get a notification. I don't get any notifications when you edit your post 😞

**Comment by mnmercer** [comment]
@dflynn15 Thank you. I look forward to the revision. 
I apologize about that. This is the first time I've really made comments on anything in GitHub and I did not realize you wouldn't get notifications.

**Comment by avitevet** [comment]
Thanks @gwynjudd , your solution fixed the problem on my PC (Win10 x64)!

**Comment by klstromsland** [comment]
Hello,
I am not sure if what I am experiencing is related to this thread.  I see that the fixes are in place in the latest version and that it is successful.  Yet I am getting the error below.  I too am going through the same coursework at Udacity.  I would very much appreciate any insight on this.   What are the possible causes?

Working from Samsung Laptop, AMD A6-3420M APU with Radeon(tm) HD Graphics 1.5GHz, 64bit, Windows 7.

Error messages:
C:\dev\mws\ud892\test_gulp>gulp tests
[17:01:19] Using gulpfile C:\dev\mws\ud892\test_gulp\gulpfile.js
[17:01:19] Starting 'tests'...
[17:01:19] Running Jasmine with PhantomJS
Couldn't load the page


[17:01:21] 'tests' errored after 2.7 s
[17:01:21] Error in plugin 'gulp-jasmine-phantomjs'
Message:
    1: Tests contained failures. Check logs for details.
Details:
    domainEmitter: [object Object]
    domain: [object Object]
    domainThrown: false

package.json
{
  "name": "test_gulp",
  "version": "1.0.0",
  "description": "",
  "main": "gulpfile.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "browser-sync": "^2.24.4",
    "gulp-autoprefixer": "^5.0.0",
    "gulp-eslint": "^4.0.2",
    "gulp-jasmine-phantom": "^3.0.0",
    "gulp-sass": "^4.0.1",
    "phantomjs": "^2.1.7"
  },
  "dependencies": {}
}

phantomjs 2.1.1 is installed in Program Files and is in PATH.

**Comment by klstromsland** [comment]
OK.  The npm installer does not yet install the most update version for me.  After making edits to local node_modules version, it does indeed work.

**Comment by sasokuncic** [comment]
Hi all,
As a participant of Udacity FEWD nanodegree I found the same problem with gulp-jasmine-phantom package on my Windows 10 laptop. Ricardo B. found the **solution** and publish it on the forum.

Use **[gulp-jasmine](https://www.npmjs.com/package/gulp-jasmine)** package instead of gulp-jasmine-phantom !!!

Additionally edit **gulpfile.js** ... _var jasmine = require('gulp-jasmine')_
